packages/admin/src/common/buttons/split/SplitButton.tsx

```diff
@@ -0,0 +1,117 @@
+import { Button, ButtonGroup, ButtonGroupProps, ClickAwayListener, Grow, MenuItem, MenuList, Paper, Popper } from "@material-ui/core";
+import * as React from "react";
+import { PropsWithChildren } from "react";
+
+import { useStoredState } from "../../../hooks/useStoredState";
+import { useThemeProps } from "./SplitButton.styles";
+import { SplitButtonContext } from "./SplitButtonContext";
+
+export interface SplitButtonProps extends ButtonGroupProps<any> {
+    selectedIndex?: number;
+    onSelectIndex?: (index: number, item: React.ReactElement) => void;
+    showSelectButton?: boolean;
+    localStorageKey?: string;
+    storage?: Storage;
+}
+
+// Based on https://material-ui.com/components/button-group/#split-button
+
+export const SplitButton = ({
+    selectedIndex,
+    onSelectIndex,
+    children,
+    showSelectButton,
+    localStorageKey,
+    storage,
+
+    ...restProps
+}: PropsWithChildren<SplitButtonProps>) => {
+    const [showSelectButtonState, setShowSelectButtonState] = React.useState<boolean | undefined>(undefined);
+
+    const childrenArray = React.Children.toArray(children);
+    const themeProps = useThemeProps();
+    const [uncontrolledSelectedIndex, setUncontrolledIndex] = useStoredState<number>(localStorageKey || false, 0, storage);
+    const _selectedIndex = selectedIndex ?? uncontrolledSelectedIndex;
+    const _onSelectIndex = onSelectIndex
+        ? onSelectIndex
+        : (index: number) => {
+              setUncontrolledIndex(index);
+          };
+
+    const [open, setOpen] = React.useState(false);
+    const anchorRef = React.useRef<HTMLDivElement>(null);
+
+    const handleMenuItemClick = (event: React.MouseEvent<HTMLLIElement, MouseEvent>, index: number, child: React.ReactElement) => {
+        _onSelectIndex(index, child);
+        setOpen(false);
+    };
```

Comment:
the "old" SplitButton did execute the action on menu select https://gitlab.vivid-planet.com/vivid/comet/-/blob/master/admin/cms-admin/src/common/SplitButton.tsx#L23

did you change that intentionally?

Reply:
yes, i intentionally removed that, because it was kind a bit too much magic for me.

readded it with the option to disable it.

Reply:
I don't think there should be an option to change this behaviour - for the sake of consistency.

I'm fine with both ways, Please talk to Mitch about what he thinks would be best here.

Reply:
@manuelblum do you have an update on this comment?

Reply:
i really missed that comment. 

I talked with Mitch about that (maybe 2 or 3 weeks ago). That's was also a reason i made this optional, because we thought that in some cases the behaviour of auto calling onClick is not intended in every situation. For sure consistency could suffer on that, but this option is still optional and the default is, that the onClick Callback is called. I think it is ok.